### PR TITLE
Prepare a new drive for shipment

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

**Comment by ppp0** [comment]
`_getStreamInput()->confirm` needs a php linked against `libreadline`, ie not a stock debian php (this is a bug/misunderstanding which will be corrected in php 5.5 afaiu)

**Comment by ppp0** [comment]
>How about instead writing a _cleanup() function, using it always at the end of each task where necessary?

Now I end up calling `_cleanup` from 7 different places and very likely I do not cover all possible edge cases (had to insert a couple of `try..catch`es just to clean up)...

**Comment by ppp0** [comment]
* `/puppet/manifest/default.pp`: https://gist.githubusercontent.com/ppp0/79473f8ede8e461a0c2b/raw/8aca8faf4bb4be9cf8fc2ffdb0095da70e63ddaf/defult.pp
* `/Vagrantfile`: https://gist.githubusercontent.com/ppp0/9ea2561065d94433c6a3/raw/cd61fae5560888c5d93fec1f6c0c547c6f47f5f1/gistfile1.txt

File: library/S3Export/library/S3Export/Device.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+
+class S3Export_Device {
+
+    /** @var string */
+    private $path;
```
Comment (tomaszdurka) [inline library/S3Export/library/S3Export/Device.php]: Rename to `_path`

**Comment by tomaszdurka** [comment]
@ppp0 Please review or call me for pair-review.

**Comment by ppp0** [comment]
* `S3Export_Device::_isPartitioned()`: this is not really a check for being partioned, but rather if the user has given a partition (containing a trailing number) or a raw device as an argument (`$devicePath`). IMO we could refactor this to `_isDevicePathAPartition` which looks ugly, unfortunately, but expresses what is. Other ideas?

**Comment by ppp0** [comment]
Paired

**Pull request merged**